multiple fixes for 11.1.5 through 11.2.7

zaphon opened the pull request:
This one is a bit more complicated than the other changes. I added a bunch of nil checking to prevent errors, but I'm afraid that it might be breaking actual functionality, so am testing, and will update as needed.

zaphon committed:
multiple fixes for 11.1.5 through 11.2.7

zaphon committed:
another nil check

zaphon committed:
more fixes

tmg-pub commented:
I'm not sure about the added nil checks. Some values are used for routing, so using something else is going to break it anyway. From what I remember, realm IDs are used to optimize the routing protocol. Potentially it could use realm names or a hash instead, but this code is fragile since there are no tests

tmg-pub commented:
Thanks - I've applied some of the safer seeming fixes selectively

zaphon commented:
I added the nil checks as I kept running into situations where the items were nil and it was throwing errors into bugsack. And what I found is as I documented, sometimes it just returns nil, and other times it doesn't So to be safe I added in checks to just bail or in some cases, find another way to get the info.

zaphon commented:
I noticed you left out the following change..

`local guid = UnitGUID( unit ) -- Fixed: was using "player" instead of unit`

Using "player" instead of unit makes no sense to me there as you're trying to get the info of unit that was passed in.
